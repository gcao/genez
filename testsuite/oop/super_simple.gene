# Test simple super method calls

# Basic super call
(class Base
  (.fn greet []
    (println "Hello from Base")))

(class Child < Base
  (.fn greet []
    (println "Hello from Child")
    (super .greet)))

(var child (new Child))
(child .greet)

# Super with different method names (no infinite loop risk)
(class Animal
  (.fn make-sound []
    (println "Generic animal sound")))

(class Dog < Animal
  (.fn bark []
    (println "Woof!")
    (super .make-sound)))

(var dog (new Dog))
(dog .bark)

# Super with return values
(class Calculator
  (.fn compute [x]
    (* x 2)))

(class AdvancedCalc < Calculator
  (.fn compute [x]
    (var base-result (super .compute x))
    (+ base-result 10)))

(var calc (new AdvancedCalc))
(println "Result:" (calc .compute 5))