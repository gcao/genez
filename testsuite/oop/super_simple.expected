Hello from Child
Hello from Base
Woof!
Generic animal sound
Result: 20